Java/app/src/main/java/com/taboola/sdk3example/sdk_native/NativeFeedAdapter.java

```diff
@@ -0,0 +1,85 @@
+package com.taboola.sdk3example.sdk_native;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.taboola.android.tblnative.TBLImageView;
+import com.taboola.android.tblnative.TBLRecommendationItem;
+import com.taboola.android.tblnative.TBLTextView;
+import com.taboola.sdk3example.R;
+
+import java.util.ArrayList;
+
+public class NativeFeedAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    private ArrayList<Object> mData;
+
+    public NativeFeedAdapter(ArrayList<Object> data) {
+        mData = data;
+    }
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View linearLayout = inflater.inflate(R.layout.feed_taboola_item, parent, false);
+        return new TaboolaItemViewHolder(linearLayout);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+        if (mData == null) {
+            return;
+        }
+
+        TBLRecommendationItem item = (TBLRecommendationItem) mData.get(position);
+        ViewGroup adContainer = ((TaboolaItemViewHolder) holder).mAdContainer;
+        Context context = adContainer.getContext();
```

Comment:
Worth explaining how Taboola  Native integration expects the publisher to implement their Taboola UI using the different Taboola views

Reply:
Done